### PR TITLE
feat: initial environment variable provider

### PR DESCRIPTION
**Pull request by beeme1mr**
## This PR

- adds a basic environment variable provider

### Notes

This provider is intended to be used in the playground and in some updated getting started documentation.

### CODE REVIEW

**Comment by beeme1mr** [comment]
Hey @toddbaert, I'm running into a build issue due to the flagd web schema git submodule not being found. Any idea how to resolve that? I don't see a reference to that in my branch. 🤷‍♂️

**Comment by moredip** [comment]
FYI, here's an even more minimalist provider that I created just for pedagogical reasons: https://github.com/moredip/openfeature-playground/tree/main/libs/openfeature-minimalist-provider

I probably needs a bit of cleanup before being used in documentation though.

**Review by moredip — APPROVED** [review]

**Comment by toddbaert** [comment]
> Hey @toddbaert, I'm running into a build issue due to the flagd web schema git submodule not being found. Any idea how to resolve that? I don't see a reference to that in my branch. man_shrugging

I would just revert your changes to the [submodule](https://github.com/open-feature/js-sdk-contrib/pull/239/files#diff-8f095f8e60c56bc4f83ef981e9560a2a9ca7c80b4e1e0619430400aeb810e43c). It seems you adding a duplicate one.

**Comment by toddbaert** [comment]
@beeme1mr I removed the submodule you inadvertently (I think) added, and rebased on main.

File: libs/providers/env-var/src/lib/env-var-provider.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+};
+
+export class EnvVarProvider implements Provider {
+  metadata = {
+    name: 'environment variable',
+  };
+
+  private readonly options: Config;
+
+  constructor(options: Partial<Config> = {}) {
```
Comment (toddbaert) [inline libs/providers/env-var/src/lib/env-var-provider.ts:30]:
```suggestion
  // the constructor is an ideal place to include provider configuration out-of-band with the OpenFeature SDK.
  constructor(options: Partial<Config> = {}) {
```

File: libs/providers/env-var/src/lib/env-var-provider.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+  async resolveStringEvaluation(flagKey: string): Promise<ResolutionDetails<string>> {
+    return this.evaluateEnvironmentVariable(flagKey, (value) => value);
+  }
+
+  async resolveNumberEvaluation(flagKey: string): Promise<ResolutionDetails<number>> {
+    return this.evaluateEnvironmentVariable(flagKey, (value) => {
+      const result = Number.parseFloat(value);
+      if (Number.isNaN(result)) {
+        throw new ParseError(`Unable to cast '${value}' to a number`);
```
Comment (toddbaert) [inline libs/providers/env-var/src/lib/env-var-provider.ts:58]:
```suggestion
        throw new ParseError(`'${value}' is not a number`);
```

File: libs/providers/env-var/src/lib/env-var-provider.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+  }
+
+  async resolveStringEvaluation(flagKey: string): Promise<ResolutionDetails<string>> {
+    return this.evaluateEnvironmentVariable(flagKey, (value) => value);
+  }
+
+  async resolveNumberEvaluation(flagKey: string): Promise<ResolutionDetails<number>> {
+    return this.evaluateEnvironmentVariable(flagKey, (value) => {
+      const result = Number.parseFloat(value);
+      if (Number.isNaN(result)) {
```
Comment (toddbaert) [inline libs/providers/env-var/src/lib/env-var-provider.ts:57]:
```suggestion
      // perform some basic validation
      if (Number.isNaN(result)) {
```

**Review by toddbaert — APPROVED** [review]
@beeme1mr Approved. I've added some gratuitous comments that might highlight some patterns we think are helpful if people use this as reference. Up to you to accept them (or not, if you think they are too much).

File: libs/providers/env-var/src/lib/env-var-provider.ts (hunk 2 above)
Comment (toddbaert) [reply libs/providers/env-var/src/lib/env-var-provider.ts:58]: I don't think there's technically any casting going on here.

File: libs/providers/env-var/src/lib/env-var-provider.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+      return {
+        value: parsedValue,
+        reason: StandardResolutionReasons.STATIC,
+      };
+    } catch (err) {
+      if (err instanceof ParseError) {
+        throw err;
+      }
+      throw new ParseError();
```
Comment (toddbaert) [inline libs/providers/env-var/src/lib/env-var-provider.ts:96]: Might want to add something here about an unknown error or something.

File: libs/providers/env-var/src/lib/env-var-provider.ts [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+
+  private evaluateEnvironmentVariable<T extends JsonValue>(
+    key: string,
+    parse: (value: string) => T
+  ): ResolutionDetails<T> {
+    const envVarKey = this.options.disableConstantCase ? key : constantCase(key);
+    const value = process.env[envVarKey];
+
+    if (!value) {
+      throw new FlagNotFoundError(`Unable to find an environment variable with the key '${envVarKey}'`);
```
Comment (toddbaert) [inline libs/providers/env-var/src/lib/env-var-provider.ts:82]:
```suggestion
      // an error thrown in a resolver method is caught by the SDK, and the default is returned
      throw new FlagNotFoundError(`Unable to find an environment variable with the key '${envVarKey}'`);
```

File: libs/providers/env-var/README.md [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+
+Required peer dependencies
+
+```
+$ npm install @openfeature/js-sdk
+```
+
+## Usage
+
+The environment variable provider uses the environment variables defined on the application to determine the value of a feature flag.
```
Comment (toddbaert) [inline libs/providers/env-var/README.md:23]:
```suggestion
The environment variable provider uses the environment variables to determine the value of a feature flag.
```

File: libs/providers/env-var/README.md [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+It supports `booleans`, `strings`, `numbers` and `objects` by attempting to cast the value of an environment variable to the requested type.
```
Comment (toddbaert) [inline libs/providers/env-var/README.md:24]:
```suggestion
It supports `booleans`, `strings`, `numbers` and `objects` by attempting to interpret the value of an environment variable to the requested type.
```

File: libs/providers/env-var/README.md [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+The fallback value will be return if the environment variable doesn't exist or the value can't be cast to the desired type.
```
Comment (toddbaert) [inline libs/providers/env-var/README.md:25]:
```suggestion
The default value will be returned if the environment variable doesn't exist or the value can't be cast to the desired type.
```

**Pull request merged**